spring-reactive-modules/spring-5-reactive-client-2/src/main/java/com/baeldung/limitrequests/client/Resilience4jRateLimit.java

```diff
@@ -0,0 +1,44 @@
+package com.baeldung.limitrequests.client;
+
+import java.time.Duration;
+
+import org.springframework.web.reactive.function.client.WebClient;
+
+import com.baeldung.limitrequests.client.utils.Client;
+import com.baeldung.limitrequests.client.utils.RandomConsumer;
+
+import io.github.resilience4j.ratelimiter.RateLimiter;
+import io.github.resilience4j.ratelimiter.RateLimiterConfig;
+import io.github.resilience4j.reactor.ratelimiter.operator.RateLimiterOperator;
+import reactor.core.publisher.Flux;
+
+public class Resilience4jRateLimit {
+
+    private Resilience4jRateLimit() {
+    }
+
+    public static Flux<Integer> fetch(WebClient client, int requests, int concurrency, int interval) {
+        RateLimiter limiter = RateLimiter.of("my-rate-limiter", RateLimiterConfig.custom()
+            .limitRefreshPeriod(Duration.ofMillis(interval))
+            .limitForPeriod(concurrency)
+            .timeoutDuration(Duration.ofMillis(interval * concurrency))
```

Comment:
This kind of operation should be casted to long before assignment, so:

`            .timeoutDuration(Duration.ofMillis((long) interval * concurrency))`